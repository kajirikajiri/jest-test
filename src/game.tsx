import React, { useState } from "react";
import { Board } from "./board";
import { NextPlayer } from "./nextPlayerTypes";
import { useXIsNext } from "./useXIsNext";
import { useWinner } from "./useWinner";
import { genMoves } from "./genMoves";
import { nextPlayer } from "./nextPlayer";
import { winner } from "./winner";

export const Game: React.FC = () => {
  const [history, setHistory] = useState<NextPlayer[][]>([Array(9).fill("")]);
  const [historyIndex, setHistoryIndex] = useState<number>(0);
  const [changingHistory, setChangingHistory] = useState<boolean>(false);

  const xIsNext: boolean = useXIsNext(historyIndex);
  const win: boolean = useWinner(history[historyIndex]);

  const updateHistory = (squares: NextPlayer[]) => {
    const newHistory: NextPlayer[][] = history.slice();
    const newHistoryIndex = historyIndex + 1;
    newHistory[newHistoryIndex] = squares;
    for (let i = newHistoryIndex + 1; i < 10; i++) {
      delete newHistory[i];
    }
    setHistory(newHistory);
    setHistoryIndex(newHistoryIndex);
    setChangingHistory(false);
  };

  const changeHistory = (i: number) => {
    setHistoryIndex(i);
    setChangingHistory(true);
  };

  const status: String = win
    ? "winner: " + winner(xIsNext)
    : "Next player: " + nextPlayer(xIsNext);

  const moves = genMoves(changeHistory, history);

  return (
    <div className="game">
      <div className="game-board">
        <Board
          win={win}
          nextPlayer={nextPlayer(xIsNext)}
          changingHistory={changingHistory}
          squares={history[historyIndex]}
          updateHistory={updateHistory}
        />
      </div>
      <div className="game-info">
        <div className="status" data-test="gameStatus">
          {status}
        </div>
        <ol>{moves}</ol>
      </div>
    </div>
  );
};
