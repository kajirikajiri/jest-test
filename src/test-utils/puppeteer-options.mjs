export const headless = {
  executablePath: "/usr/bin/chromium",
  headless: true,
  ignoreDefaultArgs: ["--disable-extensions"],
  args: [
    "--disable-gpu",
    "--disable-dev-shm-usage",
    "--disable-setuid-sandbox",
    "--no-first-run",
    "--no-sandbox",
    "--no-zygote",
    "--single-process",
  ],
};

export const nonHeadless = {
  executablePath: "/usr/bin/google-chrome",
  headless: false,
};
