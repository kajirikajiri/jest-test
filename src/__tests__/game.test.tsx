import React from "react";
import { Game } from "../game";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";

let container!: any;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

test("Game class", () => {
  const component = renderer.create(<Game />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
