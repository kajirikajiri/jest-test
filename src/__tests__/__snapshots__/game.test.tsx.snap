// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game class 1`] = `
<div
  className="game"
>
  <div
    className="game-board"
  >
    <div>
      <div
        className="status"
        data-test="gameStatus"
      >
        Next player: X
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
        <button
          className="square"
          data-test="square"
          onClick={[Function]}
        >
          
        </button>
      </div>
    </div>
  </div>
  <div
    className="game-info"
  >
    <div />
    <ol />
  </div>
</div>
`;
